packages/cozy-client/src/hooks/useQueryAll.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react'
+
+import { UseQueryReturnValue } from '../types'
+import { QueryDefinition } from '../queries/dsl'
+import useQuery from './useQuery'
+import useSafeState from './useSafeState'
+
+/**
+ * Fetches a queryDefinition and run fetchMore on the query until the query is fully loaded
+ *
+ * @param  {QueryDefinition} queryDefinition - Definition created with Q()
+ *
+ * @typedef {object} UseQueryOptions
+ * @property  {object} as - Name for the query [required]
+ * @property  {boolean} enabled - If set to false, the query won't be executed
+ * @property  {object} fetchPolicy - Fetch policy
+ * @property  {object} singleDocData - If true, the "data" returned will be
+ * a single doc instead of an array for single doc queries. Defaults to false for backward
+ * compatibility but will be set to true in the future.
+ * @property {Function} onError - Callback if the query is errored
```

Comment:
all those properties are not specific to this function. Is it possible to only inject `UseQueryOptions` and not specifiying it? (and still have the documentation generated)

Reply:
Qu'est-ce que tu entends par "pas spécifique à cette fonction" ? useQueryAll attend les mêmes param que useQuery (puisqu'il ne fait qu'un useQuery grosso modo), la JSDoc est donc la même non ?

Reply:
serait-il possible de mutualiser les paramètres dans ce cas ?

Reply:
En fait je comprends pas, `@typedef {object} UseQueryOptions` suffit non ?

Reply:
ah ok https://jsdoc.app/tags-typedef.html faudrait sortir le @typedef avec ses @property et l'utiliser en tant que @param dans useQuery et useQueryAll c'est ça ?

Reply:
J'ai mis à jour la PR en sortant les types dans le fichier approprié